File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        long ans = 0, diff = 0;
        PriorityQueue <Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int i : happiness) pq.add(i);
        while(k > 0){
            long tm = pq.remove() - diff;
            if(tm > 0) ans += tm;
            diff++;
            k--;
        }
        return ans;
    }

}

File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.js
/**
 * @param {number[]} happiness
 * @param {number} k
 * @return {number}
 */
var maximumHappinessSum = function (happiness, k) {
    happiness.sort((a,b) => b-a);
    let ans = 0, diff = 0;
    let i = 0;

    while (k > 0) {
        let tm = happiness[i] - diff;
        if (tm > 0) ans += tm;
        diff++;
        k--;
        i++;
    }
    return ans;
};